src/atc/etl/transformer.py

```diff
@@ -0,0 +1,29 @@
+from abc import abstractmethod
+
+from pyspark.sql import DataFrame
+
+
+class Transformer:
+    @abstractmethod
+    def process(self, df: DataFrame) -> DataFrame:
+        return df
+
+
+class MultiInputTransformer:
+    @abstractmethod
+    def process_many(self, dataset: {}) -> DataFrame:
+        pass
+
+
+class DelegatingTransformer(Transformer):
+    def __init__(self, inner_transformers: [Transformer]):
```

Comment:
I didn't know `[Transformer]` was allowable as a type hint. Is this equivalent to `typing.List[Transformer]`?

Reply:
I am also used to use typing.List[Transformer]

Reply:
> I didn't know `[Transformer]` was allowable as a type hint. Is this equivalent to `typing.List[Transformer]`?

I have honestly no clue. I'm still trying to learn, but there were no errors or warnings in PyCharm. It seems that the internet favors `List[T]` and if its on the internet it must be correct :D

Reply:
Done